In [2]:
!python -m pip install sklearn

Using legacy 'setup.py install' for sklearn, since package 'wheel' is not installed.
    Running setup.py install for sklearn: started
    Running setup.py install for sklearn: finished with status 'done'


In [3]:
%load_ext autoreload
%autoreload 2

In [2]:
import sklearn

In [3]:
import pandas as pd

In [5]:
!dir.\files\auto-mpg.csv

 C 드라이브의 볼륨: Windows
 볼륨 일련 번호: 729B-BAE0

 C:\Develops\multi_machinelearning\files 디렉터리

2021-06-29  오후 04:09            21,913 auto-mpg.csv
               1개 파일              21,913 바이트
               0개 디렉터리  200,379,277,312 바이트 남음


In [9]:
pd_data.shape

(398, 9)

object 주의깊게 확인하기
숫자로 들어가야하는데 object로 잡혔다면 의심해야해~
그래서 빼고 진행할게

* 데이터 읽고 필요한 부분 찾음

In [6]:
pd_data=pd.read_csv('./files/auto-mpg.csv', header=None)
pd_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       398 non-null    float64
 1   1       398 non-null    int64  
 2   2       398 non-null    float64
 3   3       398 non-null    object 
 4   4       398 non-null    float64
 5   5       398 non-null    float64
 6   6       398 non-null    int64  
 7   7       398 non-null    int64  
 8   8       398 non-null    object 
dtypes: float64(4), int64(3), object(2)
memory usage: 28.1+ KB


In [13]:
pd_data.columns = ['mpg','cylinders','displacement','horsepower','weight',
              'acceleration','model year','origin','name']
pd_data

mpg  cylinders  displacement horsepower  weight  acceleration  \
0    18.0          8         307.0      130.0  3504.0          12.0   
1    15.0          8         350.0      165.0  3693.0          11.5   
2    18.0          8         318.0      150.0  3436.0          11.0   
3    16.0          8         304.0      150.0  3433.0          12.0   
4    17.0          8         302.0      140.0  3449.0          10.5   
..    ...        ...           ...        ...     ...           ...   
393  27.0          4         140.0      86.00  2790.0          15.6   
394  44.0          4          97.0      52.00  2130.0          24.6   
395  32.0          4         135.0      84.00  2295.0          11.6   
396  28.0          4         120.0      79.00  2625.0          18.6   
397  31.0          4         119.0      82.00  2720.0          19.4   

     model year  origin                       name  
0            70       1  chevrolet chevelle malibu  
1            70       1          buick skylark 320  
2            70       1         plymouth satellite  
3            70       1              amc rebel sst  
4            70       1                ford torino  
..          ...     ...                        ...  
393          82       1            ford mustang gl  
394          82       2                  vw pickup  
395          82       1              dodge rampage  
396          82       1                ford ranger  
397          82       1                 chevy s-10  

[398 rows x 9 columns]

* 분류를 함

In [15]:
x=pd_data[['weight']]

In [16]:
y=pd_data[['mpg']]

In [17]:
x.shape , y.shape

((398, 1), (398, 1))

* 가지고 있는 데이터를 쪼개서 따로 빼놓고
  두개의 데이터를 비교해서 체크한다

In [18]:
from sklearn.linear_model import LinearRegression

In [19]:
LinearRegression()

LinearRegression()

In [20]:
lr=LinearRegression()

In [21]:
lr.fit(x,y)

LinearRegression()

In [22]:
lr.coef_

array([[-0.00767661]])

In [25]:
lr.intercept_

array([46.31736442])

패턴을 나타내는 일직선의 방정식

y=-0.00767661x + 46.31736442

* 정확도: score
* x절편: coef_
* y절편: intercept_

In [26]:
lr.score(x,y)

0.6917929800341573

-----------------------

In [27]:
pd_data[['mpg','weight']]

mpg  weight
0    18.0  3504.0
1    15.0  3693.0
2    18.0  3436.0
3    16.0  3433.0
4    17.0  3449.0
..    ...     ...
393  27.0  2790.0
394  44.0  2130.0
395  32.0  2295.0
396  28.0  2625.0
397  31.0  2720.0

[398 rows x 2 columns]

In [28]:
x=pd_data[['weight']]
y=pd_data[['mpg']]

from sklearn.model_selection import train_test_split

In [29]:
train_test_split(x,y)

[     weight
 308  2556.0
 73   4098.0
 19   1835.0
 378  2125.0
 187  4215.0
 ..      ...
 290  4054.0
 365  3060.0
 87   3988.0
 305  2670.0
 275  3140.0
 
 [298 rows x 1 columns],
      weight
 38   4209.0
 371  2525.0
 353  2190.0
 270  2515.0
 184  2572.0
 ..      ...
 250  3735.0
 316  3381.0
 350  2215.0
 242  2600.0
 72   3892.0
 
 [100 rows x 1 columns],
       mpg
 308  33.5
 73   13.0
 19   26.0
 378  38.0
 187  17.5
 ..    ...
 290  15.5
 365  20.2
 87   13.0
 305  28.4
 275  17.0
 
 [298 rows x 1 columns],
       mpg
 38   14.0
 371  29.0
 353  33.0
 270  21.1
 184  25.0
 ..    ...
 250  19.4
 316  19.1
 350  34.7
 242  21.5
 72   15.0
 
 [100 rows x 1 columns]]

In [38]:
X_train, X_test,Y_train, Y_test =train_test_split(x,y)
X_train.shape, X_test.shape,Y_train.shape, Y_test.shape

((298, 1), (100, 1), (298, 1), (100, 1))

In [39]:
from sklearn.linear_model import LinearRegression

In [40]:
lr=LinearRegression()

In [41]:
lr.fit(X_train,Y_train)

LinearRegression()

In [42]:
#total data
lr.coef_,lr.intercept_

(array([[-0.00747931]]), array([45.73803312]))

In [43]:
#split data
lr.coef_,lr.intercept_

(array([[-0.00747931]]), array([45.73803312]))